File: api/pkg/handlers/assignments.go
package handlers

import (
	"AlumniLink/api/pkg/models"
	"AlumniLink/api/pkg/stores"
	"encoding/json"
	"fmt"
	"net/http"

	"cloud.google.com/go/firestore"
	"github.com/gorilla/mux"
)

type AssignmentsHandler struct {
	Client *firestore.Client
}

func RegisterAssignmentsRoutes(router *mux.Router, h *AssignmentsHandler) {
	router.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.WriteHeader(http.StatusOK)
		w.Write([]byte("USER ENDPOINT"))
	})
	router.HandleFunc("/{groupId}/{id}", h.GetStudentAssignments).Methods("POST", "OPTIONS")
	router.HandleFunc("/createAssignment", h.CreateAssignment).Methods("POST", "OPTIONS")
}

func (h *AssignmentsHandler) GetStudentAssignments(w http.ResponseWriter, r *http.Request) {
	// Parse the request body
	vars := mux.Vars(r)
	groupId := vars["groupId"]
	id := vars["id"]

	fmt.Printf("Student id  = %s", id)
	// fmt.Println("Group ID: %s, Assignment ID: %s", groupId, id)
	assignments, err := stores.GetStudentAssignments(h.Client, groupId, id)

	if err != nil {
		http.Error(w, "Troule fetching assignments", http.StatusNotFound)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(assignments)
}

func (h *AssignmentsHandler) CreateAssignment(w http.ResponseWriter, r *http.Request) {
	var assignment models.Assignment

	if err := json.NewDecoder(r.Body).Decode(&assignment); err != nil {
		fmt.Println("err:", err)
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}
	fmt.Println("assignment", assignment)
	assignments, err := stores.CreateAssignment(r.Context(), h.Client, &assignment)

	if err != nil {
		http.Error(w, "Troule creating assignment", http.StatusNotFound)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(assignments)
}

File: portal/src/services/assignments.js
import utils from "./utils.js";

const BASE_URL = "assignments";

const getGroupAssignments = async ({mentorId, groupId}) => {
  try {
      console.log(`${BASE_URL}/${groupId}/mentor/${mentorId}`)
      const request = {
        url: `${BASE_URL}/${groupId}/mentor/${mentorId}`,
        method: 'GET',
      };
      const response = await utils.request(request)

      return response;
  } catch (error) {
      console.error("Error fetching assignments:", error);
      throw error;
  }
};

const getStudentAssignments = async ({id, groupId}) => {
  try {
      const request = {
        url: `${BASE_URL}/${groupId}/${id}`,
        method: 'GET',
      }
      const response = utils.request(request)

      return response;
  } catch (error) {
      console.error("Error fetching assignments:", error);
      throw error;
  }
};

const createStudentAssignments = async (assignmentInfo) => {
  try {
      console.log(assignmentInfo)
      const request = {
        url: `${BASE_URL}/createAssignment`,
        method: 'POST',
        body: JSON.stringify(assignmentInfo),
      }
      const response = utils.request(request)

      return response;
  } catch (error) {
      console.error("Error fetching assignments:", error);
      throw error;
  }
};

export default {
  getStudentAssignments,
  createStudentAssignments,
  getGroupAssignments
}